### ISSUE TITLE
Links to local files

### ISSUE THREAD

**Issue by Bruno-366** [manoelcampos/asciidoctor-ghpages-action#12]
I have a `README.adoc` file that links to another local adoc file `example.adoc`. In the gh-pages branch they are converted to html files as intended (`README.html` and `example.html`), although the link still refers to an adoc file `example.adoc`.

```
link:example.adoc[my local file]
```

is there anyway to automatically update all the links so that they refer to the html files instead?

**Comment by Bruno-366** [comment]
I know I can use [ifdef] preprocessors, but I don't think they would scale well. Would be awesome if all the links to local `.adoc` files got converted to `.html` files. Is there anyway to do this?

[ifdef]: https://gist.github.com/dcode/0cfbf2699a1fe9b46ff04c41721dda74#github-specific-customizations

**Comment by Bruno-366** [comment]
Ah, sorry my bad. Re-read the asciidoctor user guide and found cross references `<<example.adoc#,my local file>>`. Which solved the above problem.

Although now what happens is the `README.adoc` is being linked to from another file which "correctly" converts that link to `README.html` in the html version. The only problem is that `README.html` doesn't exist since it got renamed to `index.html`

**Comment by Bruno-366** [comment]
seems like [page-aliases] would be a nice solution, but I think that is specific to antora

[page-aliases]: https://docs.antora.org/antora/2.3/page/page-aliases/

**Comment by manoelcampos** [comment]
There is the `xref:README.adoc[My Page]`, but I suppose it works the same way as the `<<example.adoc#,my local file>>`, it's just more intuitive. I didn't know the `:page-aliases:` attribute, but I couldn't find any reference for plain asciidoc, just Antora.

I suppose that if you are linking to another README.adoc file, that one is inside a different directory. This way, you can link directly to that directory that the README/index will be loaded independently if you are inside GitHub or the GH Pages interface.

Consider the following directory tree:

* /README.adoc
  *  /docs/README.adoc

Your /README.adoc can have a link like this:

```asciidoc
link:/docs/[Docs]
```

**Comment by Bruno-366** [comment]
Hmm... sorry. Let me rephrase the second problem. I have these two files in my main branch:

* README.adoc
* file.adoc

inside file.adoc is the following content:

```
link:README.adoc[my beautiful readme file]
```

---

when asciidoctor-gh-pages-action builds the gh-pages branch it contains the following:

* index.html 
* file.html

file.html now links to README.html and not README.adoc (thanks to using `xref:` instead of `link:`).
However README.html doesn't exists since it got renamed to index.html

**Comment by manoelcampos** [comment]
In such a scenario, I think you can just use the link below inside your file.adoc:

```asciidoc
link:./[Home]
```

Try it and please let me know.

**Comment by Bruno-366** [comment]
ok cool, nice to now. I'll give it a try. Otherwise another option, I though of was to symlink (`ln -s`) readme.html to index.html instead of renaming it (`mv`)

**Comment by Bruno-366** [comment]
nope, your solution didn't work `link:./[Home]`, it probably would work for the html output since the browser recognizes that as index.html, but now the adoc files link to a directory rather than a file.

**Comment by manoelcampos** [comment]
Well, in this case, I think the patch-12 you made may be a solution.
I didn't know how git manages symlinks, but this [StackOverflow thread](https://stackoverflow.com/questions/954560/how-does-git-handle-symbolic-links) is very clear.

This way, send a PR from your patch that I merge it.
